clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -591,6 +591,8 @@ def err_decomp_decl_std_tuple_size_not_constant : Error<
   "is not a valid integral constant expression">;
 def note_in_binding_decl_init : Note<
   "in implicit initialization of binding declaration %0">;
+def err_arg_is_not_destructurable : Error<
+  "type %0 is not destructurable">;
```

Comment:
I'm not certain `destructurable` is a word users will really understand. How about: `type %0 is not valid for use as a structured binding initializer` or something along those lines?